stock_putaway_by_route/models/stock_move_line.py

```diff
@@ -9,6 +9,6 @@ class StockMoveLine(models.Model):
 
     @api.onchange("product_id", "product_uom_id")
     def onchange_product_id(self):
-        if self.rule_id.route_id:
+        if self.move_id.rule_id.route_id:
```

Comment:
@sebalix @guewen This fix should have been joined with a test in order to avoid future problems.